### PR TITLE
cjdns: luci-app-cjdns: fix up Makefiles, bump pkg vers

### PR DESCRIPTION
**Pull request by wfleurant**
Signed-off-by: William Fleurant <meshnet@protonmail.com>

### CODE REVIEW

**Comment by wfleurant** [comment]
Looking to match Makefile templates from @neheb re https://github.com/openwrt/packages/pull/9399

File: cjdns/Makefile [hunk 1]
```diff
@@ -18,26 +18,26 @@ include $(TOPDIR)/rules.mk
...
 PKG_VERSION:=20.4
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
-PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
-PKG_LICENSE:=GPL-3.0
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
```
Comment (neheb) [inline cjdns/Makefile]: Completely unneeded

File: cjdns/Makefile [hunk 2]
```diff
@@ -18,26 +18,26 @@ include $(TOPDIR)/rules.mk
...
 PKG_NAME:=cjdns
 PKG_VERSION:=20.4
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
-PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
-PKG_LICENSE:=GPL-3.0
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
```
Comment (neheb) [inline cjdns/Makefile]: Completely unneeded. The default is to generate a .xz archive, which is smaller.

File: cjdns/Makefile [hunk 3]
```diff
@@ -18,26 +18,26 @@ include $(TOPDIR)/rules.mk
...
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
+PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
+PKG_SOURCE_PROTO:=git
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_SOURCE_VERSION)
 
-include $(INCLUDE_DIR)/package.mk
+PKG_MAINTAINER:=William Fleurant <meshnet@protonmail.com>
+PKG_LICENSE:=GPL-3.0
```
Comment (neheb) [inline cjdns/Makefile]: Should be GPL-3.0-or-later

File: cjdns/Makefile [hunk 4]
```diff
@@ -18,26 +18,26 @@ include $(TOPDIR)/rules.mk
...
-PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
-PKG_LICENSE:=GPL-3.0
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
+PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
+PKG_SOURCE_PROTO:=git
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_SOURCE_VERSION)
```
Comment (neheb) [inline cjdns/Makefile]: Completely unnecessary

File: cjdns/Makefile (hunk 3 above)
Comment (neheb) [reply cjdns/Makefile]: PKG_LICENSE_FILES should be LICENSE

**Comment by neheb** [comment]
I'd do it like this:

```
PKG_SOURCE_PROTO:=git
PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
PKG_SOURCE_VERSION:=$(PKG_NAME)-v$(PKG_VERSION)
PKG_MIRROR_HASH:=xxx
```

**Comment by wfleurant** [comment]
> ```
> PKG_MIRROR_HASH:=xxx
> ```

Okay so we follow release/tag `v20.4` instead of current git hash.  

So is the mirror hash a SHA256 sum of the archive?   
`https://github.com/cjdelisle/cjdns/archive/cjdns-v20.4.tar.gz`  

it would then be this, no? `PKG_MIRROR_HASH:=e73246925954c2e2779a936af9b5b4f1def54865ea0ae37127dddc25f5617bbb
`

**Comment by jow-** [comment]
> So is the mirror hash a SHA256 sum of the archive?

No, its the sha256sum of the file that has been locally generated in your dl/

File: cjdns/Makefile (hunk 3 above)
Comment (wfleurant) [reply cjdns/Makefile]: yap

**Comment by neheb** [comment]
MIRROR_HASH can be automatically calculated with make check FIXUP=1

**Comment by wfleurant** [comment]
I'm building from `openwrt-sdk-18.06.4`  
Please take a look. PKG_MIRROR_HASH isn't being evaluated.  
PKG_BUILD_DIR is looking bad now.  
The last changes does compile (even w/o checksum)  

This'll take more time on my end, does anyone see anything?

**Comment by wfleurant** [comment]
Alright -- Hows it looking @neheb It should now attempt to pull cjdns-v20.4.tar.xz from mirrors.

File: cjdns/Makefile [hunk 5]
```diff
@@ -18,26 +18,29 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cjdns
 PKG_VERSION:=20.4
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
-PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
+PKG_SOURCE_VERSION:=cjdns-v$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-v$(PKG_VERSION).tar.xz
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)
```
Comment (neheb) [inline cjdns/Makefile]: Can't this be removed?

File: cjdns/Makefile [hunk 6]
```diff
@@ -18,26 +18,29 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cjdns
 PKG_VERSION:=20.4
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
-PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
+PKG_SOURCE_VERSION:=cjdns-v$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-v$(PKG_VERSION).tar.xz
```
Comment (neheb) [inline cjdns/Makefile]: Can't this be removed?

File: cjdns/Makefile [hunk 7]
```diff
@@ -18,26 +18,29 @@ include $(TOPDIR)/rules.mk
...
-PKG_SOURCE_VERSION:=45cdd8b3eebb18b6239feeef3b787e40d773edfb
-PKG_LICENSE:=GPL-3.0
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_SOURCE_URL:=https://github.com/cjdelisle/cjdns.git
+PKG_MIRROR_HASH:=fd694d9c33b68274647efe5081c5053068a499e2079d05434be6c7b4d18451d9
 
-include $(INCLUDE_DIR)/package.mk
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
```
Comment (neheb) [inline cjdns/Makefile]: Can't this be removed?

File: cjdns/Makefile (hunk 5 above)
Comment (wfleurant) [reply cjdns/Makefile]: killed both `pkg_source` and `pkg_source_subdir`

File: cjdns/Makefile (hunk 6 above)
Comment (wfleurant) [reply cjdns/Makefile]: killed `package_source`

File: cjdns/Makefile (hunk 7 above)
Comment (wfleurant) [reply cjdns/Makefile]: killed `pkg_build_dir` (Build/Compile now uses `$(BUILD_DIR)/$(PKG_SOURCE_VERSION)`)

**Comment by wfleurant** [comment]
@neheb okay governor, with the excess pkg variables cut how do you see it now?

**Comment by neheb** [comment]
seems fine

**Pull request merged**